addons/point_of_sale/models/pos_order.py

```diff
@@ -170,6 +170,41 @@ def _prepare_invoice(self):
             'user_id': self.env.uid,
         }
 
+    @api.model
+    def _get_account_move_line_group_data_type_key(self, data_type, values):
+        """
+        Return a tuple which will be used as a key for grouping account
+        move lines in _create_account_move_line method.
+        :param data_type: 'product', 'tax', ....
+        :param values:
+        :return: tuple() representing the data_type key
+        """
+        build_key_method_name = '_get_account_move_line_%s_key' % data_type
```

Comment:
Sorry but if the point was to potentially add new keys in a future refactoring, having one global method would have been better. You are adding 30 lines of code to avoid hardcoding 6 is just reducing the readability of the code.

This whole method probably needs refactoring (e.g. `insert_data` should not be a nested function) and we can improve the grouping keys [at that time](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).